latex/ledger-spec.tex

```diff
@@ -186,16 +208,118 @@ \section{Serialization}
+
+\begin{note}
+  We define $\fun{refund}$ by cases on whether or not
+  the refunding certificate has a corresponding
+  resource creating certificate.
+  If our rules are correct, then $\fun{refund}$
+  is only called in the case where such a matching
+  certificate exists.
+\end{note}
+
 The transition rules for unspent outputs are presented in
 Figure~\ref{fig:rules:utxo}. The states of the UTxO transition system,
 along with their types are defined in Figure~\ref{fig:defs:utxo}.
 Functions on these types are defined in Figure~\ref{fig:derived-defs:utxo}.
 
+
+
 \begin{figure*}
-  \emph{Primitive types}
+  \emph{Abstract types}
+  \begin{equation*}
+    \begin{array}{r@{~\in~}lr}
+      pc & \PrtclConsts & \text{protocol constants}
+    \end{array}
+  \end{equation*}
+  %
+  \emph{Derived types}
+  \begin{equation*}
+    \begin{array}{r@{~\in~}l@{\qquad=\qquad}r@{~\in~}lr}
+      \var{allocs}
+      & \Allocs
+      & hkeys \mapsto slot
+      & \Hash \to \Slot
+      & \text{resource allocations}
+    \end{array}
+  \end{equation*}
+  %
+  \emph{Abstract Functions}
+  \begin{equation*}
+    \begin{array}{r@{~\in~}lr}
+      \fun{d} & \PrtclConsts \to \DCert \to \Coin
+        & \text{deposit amount of a certificate}\\
+
+      \fun{decay} & \PrtclConsts \to \mathbb{N}\times\mathbb{Q}^{+}
+        & \text{decay constants}\\
+
+      \fun{dresource} & \Tx \to \powerset{(\DCertRegKey \uniondistinct \DCertRegPool)}
+        & \text{resource allocating certificates}\\
+
+      \fun{dderegister} & \Tx \to \powerset{\DCertDeRegKey}
+        & \text{resource releasing certificates}\\
+
+      \fun{dretire} & \Tx \to \powerset{\DCertRetirePool}
+        & \text{resource releasing certificates}\\
+
+      \fun{ttl} & \Tx \to \Slot
+        & \text{time to live}\\
+    \end{array}
+  \end{equation*}
+  \caption{Definitions used in Deposits}
+  \label{fig:defs:deposits}
+\end{figure*}
+
+\begin{figure}
+  \begin{align*}
+      & \fun{deposits} \in \PrtclConsts \to \Tx \to \Coin
+      & \text{total deposits for transaction} \\
+      & \fun{deposits}~{pc}~{tx} = \sum\limits_{c \in \fun{dresource}~tx} (\fun{d}~pc~c)
+      \nextdef
+      & \fun{poolAllocs} \in (\Hash \mapsto (\DCertRegPool \times \Slot)) \to \Allocs
+      & \text{pool allocations} \\
+      & \fun{poolAllocs}~\var{stpool} =
+          \{hash \mapsto slot \mid hash \mapsto (\_, slot) \in \var{stpool}\}
+      \nextdef
+      & \fun{refund} \in \PrtclConsts \to \Allocs \to \Slot \to \DCert \to \Coin
+      & \text{total refund for a certificate} \\
+      & \refund{allocs}{pc}{slot}{c} =\\
+      & \begin{cases}
+            0 & \text{if}~\cauthor c \notin allocs\\
+            \floor*{
+              \left(\fun{d}~pc~c\right) \cdot
+            \left(d_{\min}+(1-d_{\min})\cdot e^{-\lambda\cdot\delta}\right)}
+            & \text{otherwise}
+        \end{cases}\\
+      & \where
+        \begin{array}{r@{~=~}l}
+          \fun{decay}~pc & d_{\min},~\lambda\\
```

Comment:
where is `decay pc` used ?

Reply:
in the `where` clause, to get the price of the cert deposit and the decay constant.